### PR TITLE
LPS-132406 Keeping the side panel closed when logging in even if it was left opened before the last logout

### PR DESCRIPTION
**Pull request by YurenaC**
- LPS-132406 Create LogoutPreAction hook for A/B Testing Panel - A/B Testing Panel state will be set to closed when the user logged out
- LPS-132406 Create LogoutPreAction hook for Content Performace Panel - Content Performance Panel state will be set to closed when the user logged out

### CODE REVIEW

**Comment by YurenaC** [comment]
ci:forward

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:forward

File: modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/product/navigation/control/menu/SegmentsExperimentProductNavigationControlMenuEntry.java [hunk 1]
```diff
@@ -342,6 +337,9 @@ private void _processBodyBottomTagBody(PageContext pageContext) {
 	private static final String _ICON_TMPL_CONTENT = StringUtil.read(
 		SegmentsExperimentProductNavigationControlMenuEntry.class, "icon.tmpl");
 
+	private static final String _PANEL_STATE_SESSION_KEY =
```
Comment (4lejandrito) [inline modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/product/navigation/control/menu/SegmentsExperimentProductNavigationControlMenuEntry.java:340]: @YurenaC Brian changed the name of the key in https://github.com/brianchandotcom/liferay-portal/commit/b32e51e76ab7ca23e7073d8c493567fbf8394f24. We should call it the same.

**Review by 4lejandrito — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/product/navigation/control/menu/SegmentsExperimentProductNavigationControlMenuEntry.java (hunk 1 above)
Comment (YurenaC) [reply modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/product/navigation/control/menu/SegmentsExperimentProductNavigationControlMenuEntry.java:340]: Absolutely right, on my way!

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/product/navigation/control/menu/AnalyticsReportsProductNavigationControlMenuEntry.java [hunk 2]
```diff
@@ -308,6 +315,9 @@ private void _processBodyBottomTagBody(PageContext pageContext) {
 	private static final String _ICON_TMPL_CONTENT = StringUtil.read(
 		AnalyticsReportsProductNavigationControlMenuEntry.class, "icon.tmpl");
 
+	private static final String _SESSION_CLICKS_KEY =
```
Comment (4lejandrito) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/product/navigation/control/menu/AnalyticsReportsProductNavigationControlMenuEntry.java:318]: This is not being reused inside the `isPanelStateOpen` method.

File: modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/product/navigation/control/menu/SegmentsExperimentProductNavigationControlMenuEntry.java [hunk 3]
```diff
@@ -193,23 +192,12 @@ public boolean includeIcon(
...
 		String segmentsExperimentPanelState = SessionClicks.get(
-			httpServletRequest,
-			"com.liferay.segments.experiment.web_panelState", "closed");
+			httpServletRequest, _SESSION_CLICKS_KEY, "closed");
 
 		if (Objects.equals(segmentsExperimentPanelState, "open")) {
 			return true;
 		}
 
-		HttpServletRequest originalHttpServletRequest =
```
Comment (4lejandrito) [inline modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/product/navigation/control/menu/SegmentsExperimentProductNavigationControlMenuEntry.java:203]: Why do we have to remove this piece of code?

**Review by 4lejandrito — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/product/navigation/control/menu/SegmentsExperimentProductNavigationControlMenuEntry.java (hunk 3 above)
Comment (4lejandrito) [reply modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/product/navigation/control/menu/SegmentsExperimentProductNavigationControlMenuEntry.java:203]: I think this cannot be removed. It purpose is to open the panel when clicking on a notification. See https://github.com/liferay/liferay-portal/blob/f612f53a220b02c9ea534f4272947e7bb2b93df1/modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/notifications/SegmentsExperimentUserNotificationHandler.java#L162-L163.

**Review by 4lejandrito — APPROVED** [review]

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
ci:forward

**Pull request closed without merging**